sherpa/astro/models/tests/test_astro_model.py

```diff
@@ -95,7 +95,6 @@ def test_send_keyword_1d(cls):
     """What happens if we use an un-supported keyword?"""
 
     mdl = cls()
-    mdl._use_caching = False
```

Comment:
So, below we call

```python
    y1 = mdl(x)
    y2 = mdl(x, not_an_argument=True)
```

I forget now what happens in the code to "throw-away" this extra argument, but it's not obvious to me whether here we should still ensure there's no cache to make sure we get to the "call the model" rather than the cache just going "oh, we don't care about this argument" - e.g. instead of removing this line we should replace it with

```python
mdl.cache = 0
```

Similar for `test_send_keyword_2d` below.

Maybe we should add a parametrized argument like

```python
@pytest.mark.parametrize("cache_size", [0, 5])
```

and set

```python
mdl.cache = cache_size
```

just to make sure we check both cases without having to worry about the internals of what the cache does.

Reply:
We can make the test even more stringent. If the model is called with the extra parameter, there needs to be a second entry in the cache because the cache hash holds a different hash for the second call than the first. So, what's in there right now and what you suggest are really two different tests. Will split.